### PR TITLE
docs(research): add zoekt trigram matching analysis

### PR DESCRIPTION
**Pull request by athola**
Research for #24: Evaluated zoekt and Rust alternatives for trigram-based skill matching. Recommends `trigram` crate for fuzzy similarity matching in the intelligence crate.

Key findings:
- Zoekt has no Rust bindings, would require FFI/subprocess
- `trigram` crate provides pg_trgm-compatible similarity
- Simple API integrates cleanly with existing code
- Performant for our scale (< 100 skills typically)

Implementation plan created for follow-up in #66.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/server/src/app/intelligence.rs [hunk 1]
```diff
@@ -752,6 +752,105 @@ impl SkillService {
...
+
+        let limit = args.get("limit").and_then(|v| v.as_u64()).unwrap_or(10) as usize;
+
+        // Note: include_description is accepted but currently unused since
+        // SkillMeta doesn't cache parsed descriptions. Future enhancement could
+        // cache frontmatter descriptions for richer matching.
+        let _include_description = args
+            .get("include_description")
+            .and_then(|v| v.as_bool())
+            .unwrap_or(true);
```
Comment (athola) [inline crates/server/src/app/intelligence.rs]: **[S1] Unused parameter accepted**

The `include_description` parameter is accepted but unused (assigned to `_include_description`).

**Current behavior:** The parameter is documented in the tool schema but has no effect.

**Suggestion:** Either:
1. Remove from schema until implemented
2. Add TODO comment explaining the plan
3. Implement description caching now

**Severity:** SUGGESTION - Not blocking.

**Review by athola — COMMENTED** [review]
## PR Review Summary

### Scope Compliance: ✅ PASS

**Source:** PR description + issue #24 requirements

| Requirement | Status |
|-------------|--------|
| Research zoekt trigram matching | ✅ Complete |
| Evaluate Rust alternatives | ✅ `trigram` crate selected |
| Implement similarity module | ✅ `similarity.rs` added |
| Add MCP tool | ✅ `search-skills-fuzzy` |
| Integration tests | ✅ Comprehensive coverage |
| Version bump | ✅ 0.4.2 |

### Code Quality Analysis

| Check | Result |
|-------|--------|
| **Tests** | ✅ All passing |
| **Clippy** | ✅ No warnings |
| **Formatting** | ✅ Clean |
| **Documentation** | ✅ Research doc + inline docs |

### Blocking Issues: 0

No blocking issues found.

### In-Scope Issues: 0

All requirements implemented correctly.

### Suggestions: 1

| ID | File | Description |
|----|------|-------------|
| S1 | `intelligence.rs:783` | `include_description` parameter accepted but unused |

See inline comment for details. This is minor - the code correctly uses `_` prefix to silence warnings.

### Technical Notes

**Algorithm choice is sound:**
- `trigram` crate provides pg_trgm-compatible matching
- O(n+m) complexity per comparison is fine for <100 skills
- Case-insensitive matching built-in
- No external dependencies

**Test coverage is comprehensive:**
- Edge cases (empty strings, unicode, punctuation)
- Typo tolerance verified
- Threshold filtering tested
- Integration tests for MCP tool

### Recommendation

**APPROVE** - Implementation is clean, well-tested, and meets all requirements from issue #24.

---
*Review generated by Claude Code PR Review*

File: crates/server/src/app/intelligence.rs (hunk 1 above)
Comment (athola) [reply crates/server/src/app/intelligence.rs]: Fixed in e3480d6 - Added debug tracing when `include_description` is provided, logging that it is not yet implemented. Also updated the tool schema description to clarify this limitation.

File: crates/server/src/app/intelligence.rs [hunk 2]
```diff
@@ -752,6 +752,130 @@ impl SkillService {
...
+            .unwrap_or(DEFAULT_THRESHOLD);
+
+        // Validate and clamp limit to reasonable bounds [1, 1000]
+        let limit = args
+            .get("limit")
+            .and_then(|v| v.as_u64())
+            .map(|l| l.clamp(1, 1000) as usize)
+            .unwrap_or(10);
+
+        // Note: include_description is accepted but currently unused since
```
Comment (athola) [inline crates/server/src/app/intelligence.rs]: log this as an issue todo in this repo

File: docs/research/zoekt-trigram-research.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
+# Research: Zoekt Trigram Matching for Skrills
```
Comment (athola) [inline docs/research/zoekt-trigram-research.md]: this is good, but let's instead move these research findings and the actual implementation as an ADR

File: crates/server/src/app/intelligence.rs (hunk 2 above)
Comment (athola) [reply crates/server/src/app/intelligence.rs]: Created GitHub issue #68 to track this enhancement: https://github.com/athola/skrills/issues/68

The issue documents:
- The current limitation (description caching not implemented)
- Proposed implementation approach
- Acceptance criteria for the feature

File: docs/research/zoekt-trigram-research.md (hunk 3 above)
Comment (athola) [reply docs/research/zoekt-trigram-research.md]: Converted to ADR format in commit 49f5231:

- Created `docs/adr/0005-trigram-fuzzy-matching.md`
- Removed `docs/research/zoekt-trigram-research.md`

The ADR follows the established format and includes:
- Context and problem statement
- Decision and rationale
- Alternatives considered
- Consequences (positive/negative)
- Implementation status
- References to related issues
